src/ArrayValueCalculator.java

```diff
@@ -0,0 +1,30 @@
+import exception.ArrayDataException;
+import exception.ArraySizeException;
+
+public class ArrayValueCalculator {
+    private static final int ARRAY_ROW_NUMBER = 4;
+    private static final int ARRAY_COLUMN_NUMBER = 4;
+
+    public int doCalc(String[][] array) throws ArraySizeException, ArrayDataException {
```

Comment:
`throws ArraySizeException, ArrayDataException` should be used if you wanna delegate error handling to the end user. The description asks you to validate the input on your own, so `throws` is redundant here, remove it.